### PR TITLE
Make machine casings use ResourceLocations and translations

### PR DESCRIPTION
**Pull request by Swedz**

### CODE REVIEW

File: src/client/java/aztech/modern_industrialization/machines/models/MachineBakedModel.java [hunk 1]
```diff
@@ -57,22 +57,25 @@ public class MachineBakedModel implements IDynamicBakedModel {
     public static final String CASING_FOLDER = "machine_casing";
 
     public static ModelResourceLocation getCasingModelId(MachineCasing casing) {
-        return ModelResourceLocation.standalone(MI.id(CASING_FOLDER + "/" + casing.name));
+        return ModelResourceLocation
```
Comment (Technici4n) [inline src/client/java/aztech/modern_industrialization/machines/models/MachineBakedModel.java]: Looks like a good use case for `casing.key.withPath(p -> CASING_FOLDER + "/" + p)`.

File: docs/ADDING_MACHINES.md [hunk 2]
```diff
@@ -327,12 +327,12 @@ remember that the top, side and bottom textures of a casing must be `modern_indu
 For example:
 ```js
 MIMachineEvents.registerCasings(event => {
-    // Register two casings.
+    // Register a casing.
     // This doesn't register any model! Either add models or add the top/side/bottom textures.
-    event.register("my_fancy_casing", "my_other_casing");
+    event.register("fancy_casing", "Fancy");
```
Comment (Technici4n) [inline docs/ADDING_MACHINES.md]: This is a breaking change for KubeJS users, might be best to avoid it?

File: docs/ADDING_MACHINES.md [hunk 3]
```diff
@@ -327,12 +327,12 @@ remember that the top, side and bottom textures of a casing must be `modern_indu
...
 
     // This registers a new casing with the same model as a diamond block!
-    event.registerBlockImitation("my_diamond_casing", "minecraft:diamond_block");
+    event.registerBlockImitation("diamond_casing", "Diamond", "minecraft:diamond_block");
```
Comment (Technici4n) [inline docs/ADDING_MACHINES.md]: We can use the display name of the block in this case, no?

File: src/client/java/aztech/modern_industrialization/machines/models/MachineModelBaker.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package aztech.modern_industrialization.machines.models;
+
+import java.util.Map;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.resources.ResourceLocation;
+
+public interface MachineModelBaker {
```
Comment (Technici4n) [inline src/client/java/aztech/modern_industrialization/machines/models/MachineModelBaker.java]: All of this stuff looks unnecessary.

**Review by Technici4n — CHANGES_REQUESTED** [review]

File: docs/ADDING_MACHINES.md (hunk 3 above)
Comment (Kazuhiko-Gushiken) [reply docs/ADDING_MACHINES.md]: I agree with this, it threw off my casing code first time around.

File: docs/ADDING_MACHINES.md (hunk 2 above)
Comment (Swedz) [reply docs/ADDING_MACHINES.md]: Not sure what would be a good solution for this, the translation name for casings are used in the tooltips of blocks in the structure PR I've been working on.

File: docs/ADDING_MACHINES.md (hunk 3 above)
Comment (Swedz) [reply docs/ADDING_MACHINES.md]: I'm not seeing a clear solution for this. The english name of the block would need to be pulled from somewhere for datagen. Not sure if that is guaranteed available, especially on a server.

**Review by Technici4n — APPROVED** [review]

**Pull request merged**